a10_octavia/controller/worker/controller_worker.py

```diff
@@ -694,3 +791,39 @@ def amphora_cert_rotation(self, amphora_id):
             operator_fault_string='This provider does not support rotating '
                                   'Amphora certs. We will use preconfigured '
                                   'devices.')
+
+    def validate(self, vthunder_conf):
```

Comment:
If it's not valid for these to be None then they need to be positional arguments in the vThunder data model.

The only thing that needs validating is the ip address.